doc/BUILD.rst

```diff
@@ -0,0 +1,500 @@
+-------------
+
+System/Vendored Dependencies (``--wrap-mode``)
+""""""""""""""""""""""""""""""""""""""""""""""
+
+See: `Meson Manual <https://mesonbuild.com/Wrap-dependency-system-manual.html>`__
+
+* Default: ``default``
+* Options: ``default``, ``nofallback``, ``forcefallback``, ...
+
+This is a core ``meson`` flag that does quite a few things. Not all of them will
+be covered here. Refer to the ``meson`` documentation linked above.
+
+Generally, ``default`` will rely on system-installed libraries when available,
+and fallback to vendored in-repository dependencies when necessary.
+
+``forcefallback`` will heavily encourage the use of in-repository dependencies
+whenever possible. Recommended for release builds.
+
+``nofallback`` discourages the use of in-repository dependencies whenever
+possible, instead relying on system libraries. Useful for CI.
+
+.. code:: sh
+
+   # for release builds
+   meson configure build/ --wrap-mode=forcefallback
+   # useful for testing/CI
+   meson configure build/ --wrap-mode=nofallback
+
+Relative Directory Install (``-Dinstall_relative``)
+"""""""""""""""""""""""""""""""""""""""""""""""""""
+
+* TODO
+
+Package Data/ZIP (``-Denable_zip``/``-Dpackage_data``)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+* TODO
+
+Strict Compiler Warnings (``-Dwerror``)
+"""""""""""""""""""""""""""""""""""""""
+
+* Default: ``false``
+* Options: ``true``, ``false``
+
+This option forces stricter checks against Taisei's codebase to ensure code
+health, treating all ``Warnings`` as ``Errors`` in the code.
+
+It's highly recommended to enable this whenever developing for the engine.
+Sometimes, it's overly-pedantic, but much of the time, it provides useful
+advice. (For example, it can detect potential null-pointer exceptions that may
+not be obvious to the human eye.)
+
+.. code:: sh
+
+   meson configure build/ -Dwerror=true
+
+Deprecation Warnings (``-Ddeprecation_warnings``)
+"""""""""""""""""""""""""""""""""""""""""""""""""
+
+* Default: ``(null)``
+* Options: ``error``, ``no-error``, ``ignore``
+
+Sets deprecation warnings to either hard-fail (``error``), print as warnings but
+not trigger full errors if ``-Dwerror=true`` (``no-error``), and otherwise
+ignore them (``ignore``).
+
+Generally, ``no-error`` is the recommended default when using ``-Dwerror=true``.
+
+.. code:: sh
+
+   meson configure build/ -Ddeprecation_warnings=no-error
+
+
+In-Game Developer Options (``-Ddeveloper``)
+"""""""""""""""""""""""""""""""""""""""""""
+
+* Default: ``false``
+* Options: ``true``, ``false``
+
+For testing actual gameplay, you can set this option and it will enable cheats
+and other 'fast-forward' options by the pressing keys defined in
+``src/config.h``.
+
+.. code:: sh
+
+   meson configure build/ -Ddeveloper=true
+
+Build Type (``-Dbuildtype``)
+""""""""""""""""""""""""""""
+
+* Default: ``release``
+* Options: ``release``, ``debug``
+
+Sets the type of build. ``debug`` enables several additional debugging features.
```

Comment:
Several additional debugging features means reduced optimizations and debug symbols right? Would be nice to clarify the difference to `-Ddeveloper`.